### PR TITLE
MDS-2884: Batch Downloading of NoW Docs

### PR DESCRIPTION
**Pull request by BenHarrington**
# Main

- Ticket: https://bcmines.atlassian.net/jira/software/c/projects/MDS/boards/1?modal=detail&selectedIssue=MDS-2884
- Add new "Download Documents" button on Manage Documents tab, right of the tab header
- Create new modal based off of the Referral/Consultation package download package modal
- Fixed progress bar to update correctly as files get downloaded

# Other

- Disregard docgen commits in this ticket, those will be resolved before this gets merged in

# How to test

- Visit a NoW that has the "Manage Documents" tab available
- Click on the "Download Documents" button to the right of the tab header
Core-manage-documents
- Verify all documents from all sections of that tab show up in the modal
download-now-documents
- Click "Download Selected Documents" and ensure the progress widget updates accordingly and that the success toast upon completion reflects the correct number of files downloaded

### CODE REVIEW

File: services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js [hunk 1]
```diff
@@ -58,6 +67,121 @@ export class ManageDocumentsTab extends Component {
...
+        });
+        this.downloadDocument(url);
+        // eslint-disable-next-line
+        await new Promise((resolve) => setTimeout(resolve, 2000));
+      }
+      // dispatch toast message
+      notification.success({
+        message: `Successfully Downloaded: ${totalFiles} files.`,
+        duration: 10,
+      });
```
Comment (areyeslo) [inline services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js:138]: Would be possible to decouple the logic and put it in components/common folder to be reused in any other parts of Core?

**Comment by JustinMacaulay** [comment]
I would appreciate one other set of eyes on this but happy to approve after that if there are no major requests.

**Review by areyeslo — COMMENTED** [review]
Looks good to me. Great Job on this Ben.

File: services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js (hunk 1 above)
Comment (BenHarrington) [reply services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js:138]: There are already 3 different places this occurs, I believe one was busted. Which part of the code in particular would you like to extract?

**Review by JustinMacaulay — APPROVED** [review]
Looks good.

**Pull request merged**